lib/rubocop/config_loader_resolver.rb

```diff
@@ -101,6 +103,32 @@ def duplicate_setting?(base_hash, derived_hash, key, inherited_file)
       Gem.path.none? { |dir| inherited_file.start_with?(dir) } # Can change?
     end
 
+    def warn_on_duplicate_setting(base_hash, derived_hash, key, **opts)
+      return unless duplicate_setting?(base_hash, derived_hash,
+                                       key, opts[:inherited_file])
+
+      inherit_mode = opts[:inherit_mode] && opts[:inherit_mode]['merge']
+      return if base_hash[key].is_a?(Array) &&
+                inherit_mode && inherit_mode.include?(key)
+
+      warn("#{PathUtil.smart_path(opts[:file])}: " \
+           "#{opts[:cop_name]}:#{key} overrides " \
+           "the same parameter in #{opts[:inherited_file]}")
+    end
+
+    def determine_inherit_mode(hash, key)
+      cop_cfg = hash[key]
+      local_inherit = cop_cfg.delete('inherit_mode') if cop_cfg.is_a?(Hash)
```

Comment:
Why do we need `delete` here? All specs still pass if I change it to `cop_cfg['inherit_mode']`.

Reply:
That makes sense. It's not necessary to implement `merge_with_default` at this point IMO. We can do one use case at a a time.

Thanks for the added spec example.